huxley/www/js/components/ChairAttendanceView.js

```diff
@@ -67,50 +100,59 @@ var ChairAttendanceView = React.createClass({
-    //   return (
-    //     <tr>
-    //       <td>
-    //         {country.name}
-    //       </td>
-    //       <td>
-    //           <label name="committee_prefs">
-    //             <input
-    //               className="choice"
-    //               type="checkbox"
-    //               name="committee_prefs"
-    //             />
-    //           </label>
-    //       </td>
-    //       <td>
-    //           <label name="committee_prefs">
-    //             <input
-    //               className="choice"
-    //               type="checkbox"
-    //               name="committee_prefs"
-    //             />
-    //           </label>
-    //       </td>
-    //       <td>
-    //           <label name="committee_prefs">
-    //             <input
-    //               className="choice"
-    //               type="checkbox"
-    //               name="committee_prefs"
-    //             />
-    //           </label>
-    //       </td>
-    //     </tr>
-    //   )}.bind(this));
-    return (
-      <tr></tr>
+    var committeeCountryIDs = Object.keys(this.state.country_assignments);
+    var countries = this.state.countries;
+    return committeeCountryIDs.map(country => 
+      <DelegationAttendanceRow
+        key={country}
+        onChange={this._handleAttendanceChange}
+        countryName={Object.keys(countries).length ? countries[country].name : country}
+        countryID={country}
+        delegates={this.state.country_assignments[country]}
+      />
     );
   },
 
+  _mapAssignments() {
+    var user = CurrentUserStore.getCurrentUser();
+    var country_assignments = {};
+    var delegates = this.state.delegates;
+    var assignments = this.state.assignments;
+    for (var delegate of delegates) {
+      var assignment = assignments.find(assignment => assignment.id == delegate.assignment);
+      if (!assignment) continue;
+      var countryID = assignment.country;
+      if (countryID in country_assignments) {
+        country_assignments[countryID].push(delegate);
+      } else {
+        country_assignments[countryID] = [delegate];
+      }
+    }
+    this.setState({country_assignments: country_assignments});
+  }, 
+
+  _handleAttendanceChange(session, country, event) {
+    var country_assignments = this.state.country_assignments;
+    var country_delegates = country_assignments[country];
+    for (var delegate of country_delegates) {
+      delegate[session] = !delegate[session];
+    }
+
+    this.setState({country_assignments: country_assignments});
+  },
+
+  _handleSaveAttendance(event) {
+    var committee = CurrentUserStore.getCurrentUser().committee;
+    var country_assignments = this.state.country_assignments;
+    var delegates = [];
+    for (var country in country_assignments) {
+      delegates = delegates.concat(country_assignments[country]);
+    }
+    DelegateActions.updateCommitteeDelegates(committee, delegates);
+    window.alert("Attendance Saved.");
```

Comment:
Probably not worth throwing this up in case an error is thrown by the server. We can handle telling advisors their save was successful in a separate diff.